pkg/user/user_integration_test.go

```diff
@@ -0,0 +1,187 @@
+package user_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/dhis2-sre/im-manager/internal/middleware"
+	"github.com/dhis2-sre/im-manager/pkg/group"
+	"github.com/dhis2-sre/im-manager/pkg/inttest"
+	"github.com/dhis2-sre/im-manager/pkg/model"
+	"github.com/dhis2-sre/im-manager/pkg/token"
+	"github.com/dhis2-sre/im-manager/pkg/user"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserHandler(t *testing.T) {
+	t.Parallel()
+
+	db := inttest.SetupDB(t)
+	userRepository := user.NewRepository(db)
+	userService := user.NewService(userRepository)
+	groupRepository := group.NewRepository(db)
+	groupService := group.NewService(groupRepository, userService)
+
+	err := createAdminUser("admin", "admin", userService, groupService)
+	require.NoError(t, err, "failed to create admin user and group")
+
+	authorization := middleware.NewAuthorization(userService)
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err, "failed to generate private key")
+	// TODO(ivo) I don't think we need/should use a pointer
```

Comment:
Please elaborate :smile:

Reply:
The middleware is not and should not be mutating the certificate. Pointers convey the opposite 😄